File: app/src/main/java/ru/eva/oriokslive/activities/Events/EventsActivity.java
package ru.eva.oriokslive.activities.Events;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import java.util.List;

import ru.eva.oriokslive.R;
import ru.eva.oriokslive.adapters.EventsAdapter;
import ru.eva.oriokslive.helpers.DialogHelper;
import ru.eva.oriokslive.models.orioks.Disciplines;
import ru.eva.oriokslive.models.orioks.Events;


public class EventsActivity extends AppCompatActivity implements Contract.View {

    private EventsAdapter adapter;
    private Contract.Presenter mPresenter;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_events);

        Intent intent = getIntent();
        String intExtraName = "id";
        int id = intent.getIntExtra(intExtraName, 0);

        mPresenter = new Presenter(this, id);
        mPresenter.getToolbarTitle();
        mPresenter.getRecyclerView();
        mPresenter.getEvents(this);
    }



    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.events, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_view:
                mPresenter.optionsItemSelected();
                break;
            default:
                onBackPressed();
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void setRecyclerView(List<Events> data) {
        RecyclerView recyclerView = findViewById(R.id.recycler_view);
        adapter = new EventsAdapter(data);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void showDialog(Disciplines discipline, DialogHelper dialog) {
        dialog.createDialog(discipline, this).show();
    }

    @Override
    public void setToolbarTitle(String title) {
        Toolbar toolbar = findViewById(R.id.toolbar);
        toolbar.setTitle(title);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public void addRecyclerViewItems(List<Events> data) {
        adapter.addItems(data);
    }

    @Override
    public void showToast(String text) {
        Toast.makeText(this, text, Toast.LENGTH_LONG).show();
    }
}

File: app/src/main/java/ru/eva/oriokslive/ui/activity/events/EventsActivity.kt
package ru.eva.oriokslive.ui.activity.events

import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import androidx.activity.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import dagger.hilt.android.AndroidEntryPoint
import ru.eva.oriokslive.R
import ru.eva.oriokslive.databinding.ActivityEventsBinding
import ru.eva.oriokslive.network.exceptions.NetworkException
import ru.eva.oriokslive.ui.adapter.EventsAdapter
import ru.eva.oriokslive.ui.base.BaseActivity
import ru.eva.oriokslive.ui.dialog.DisciplineDialog
import ru.eva.oriokslive.utils.showToast

@AndroidEntryPoint
class EventsActivity : BaseActivity<ActivityEventsBinding>() {

    override val bindingInflater: (LayoutInflater) -> ActivityEventsBinding =
        ActivityEventsBinding::inflate
    private val viewModel: EventsViewModel by viewModels()

    private val id: Int by lazy { intent.getIntExtra(EXTRA_ID, 0) }

    override fun setupUI() {
        viewModel.getTitle(id)
        viewModel.getEvents(id)

        viewModel.title.observe(this) {
            title = it
            setSupportActionBar(binding.toolbar)
            supportActionBar?.apply {
                setDisplayHomeAsUpEnabled(true)
                setDisplayShowHomeEnabled(true)
            }
        }

        viewModel.events.observe(this) {
            binding.rvEvents.layoutManager = LinearLayoutManager(this)
            binding.rvEvents.adapter = EventsAdapter(it)
        }

        viewModel.discipline.observe(this) {
            DisciplineDialog(this, it).show()
        }
        viewModel.onError.observe(this) {
            if (it is NetworkException) viewModel.getLocalEvent(id)
            showToast(it)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.events, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
            R.id.actionView -> viewModel.getDiscipline(id)
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        const val EXTRA_ID = "ID"
    }
}